docs/working-with-projects/working-with-images/sign-images.md

```diff
@@ -19,6 +19,7 @@ A key feature of using Cosign with Harbor is the ability use Harbor's [replicati
 
 * When replicating from Harbor to another target registry type, the target registry will not manage the link between the signed artifact and any associated signatures. You will see the subject manifest and signatures as coordinating artifacts under the same repository.
 
+**Note:** Only Manual and Scheduled replications "Trigger Modes" are applicable. Event-based replication is not possible at the moment due to the chicken and egg situation, you cannot replication image which is not signed, but you cannot replicate signature without image.
```

Comment:
Cosign signature is actually a OCI compatible artifact. But, from the perspective of Harbor, when it recevies the signature push request, it will check the existence of the subject manifest, which the signature belongs to. If the subject manifest doesn't exist, harbor will reject the signature push request via 404.

On top of that, if you enable the event based replication, you have to guarantee that the subject manifest has already be replicated to the target Harbor, otherwise the replicaiton that triggered by the push event of signature will be failed by the above reason.

The event based has limition on the following scenario, besides that it works fine. 

Let's give one typcial example here, it assumes that we have two Harbor instances, A and B.

- step 1, user pushes imageA into Harbor A.
- step 2, user enables the event base replication from A to B. At this time, the imageA doesn't exist in the target Harbor B.
- step 3, user signes the imageA via cosign. The Harbor A receive the event on pushing signature, then it triggers a replicaiton to replicate the signature from Harbor A to B.
- step 4, user see the 404 error. 

But, if we revise the steps to:
- step 1, user enables the event base replication from Harbor A to B. 
- step 2, user pushes imageA into Harbor A. Harbor triggeres a replcaiton to replcate the imageA from Harbor A to B.
- step 3, user signes the imageA via cosign. Harbor triggeres a replcaiton to replcate the signature from Harbor A to B.

Everything works.

Above is the currently design, and the [open issue](https://github.com/goharbor/harbor/issues/16461) on it.

Reply:
so do you want to change the docs? or to leave it as it is?

Reply:
Just want to give more explain to make sure we're on the same page.

Im ok to leave it as it is.